core/src/test/java/org/apache/iceberg/TestPartitionSpecUpdate.java

```diff
@@ -0,0 +1,255 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+@RunWith(Parameterized.class)
+public class TestPartitionSpecUpdate extends TableTestBase {
```

Comment:
I don't think that this test should extend `TableTestBase`. Instead, I think it should be like `TestSchemaUpdate`, which instantiates `SchemaUpdate` directly by passing in the schema to update and the highest assigned ID. We don't need to test the commit mechanism, just how updates are applied to the base `PartitionSpec`. That would also make it easier to handle v1 and v2 tests, instead of using `V1Assert` and `V2Assert`.

Reply:
This would also remove the need to make assertions about the table's current spec in every test case. Instead, you'd just pass in the spec you want to make changes to.

Reply:
I moved some tests to a new test `TestTablePartitionSpecUpdate`, where it needs the table for tests related to table metadata, e.g. version, spec history, specId, etc.